Archives/VolFile.cpp

```diff
@@ -114,12 +128,14 @@ namespace Archives
 	{
 		try
 		{
-			char* offset = (char*)m_IndexEntries[fileIndex].dataBlockOffset;
-			int length = *(int*)(offset + 4) & 0x7FFFFFFF;
-			offset += 8;
+			SectionHeader sectionHeader = GetSectionHeader(fileIndex);
 			
+			std::vector<void*> packedFileBuffer;
+			packedFileBuffer.resize(sectionHeader.length);
+			archiveFileReader->Read(packedFileBuffer.data(), sectionHeader.length);
+
```

Comment:
This should be refactored to read files in a standard chunk size instead of whole files to improve performance on very large files. Maybe use the constant VOL_WRITE_SIZE?

Reply:
You're allocating 4x-8x more memory than you probably think you are ;)

Agreed on using a chunked write. It's a fairly generic operation to copy from one stream to another. Perhaps this can be abstracted to a utility function. For now though, this should be sufficient to fix the current issues. Optimization can always be done later.

The `void*` should of course be changed to a `uint8_t` or `char`-type.